client/mm/libxc/coinbase.go

```diff
@@ -0,0 +1,1670 @@
+	c.booksMtx.RUnlock()
+	if !found {
+		c.log.Errorf("no book found for %s", productID)
+		return 0
+	}
+
+	midGap, err := book.midGap()
+	if err != nil {
+		c.log.Errorf("error getting mid gap: %v", err)
+		return 0
+	}
+
+	return midGap
+}
+
+// GetDepositAddress returns a deposit address for the specified asset.
+func (c *coinbase) GetDepositAddress(ctx context.Context, assetID uint32) (string, error) {
+	ticker, found := c.idTicker[assetID]
+	if !found {
+		return "", fmt.Errorf("no ticker found for asset ID %d", assetID)
+	}
+
+	acctID, err := c.getAccountID(ctx, ticker)
+	if err != nil {
+		return "", err
+	}
+
+	endpoint := fmt.Sprintf("v2/accounts/%s/addresses", acctID)
+	var res cbtypes.DepositAddressResponse
+	if err := c.request(ctx, http.MethodPost, endpoint, nil, nil, &res); err != nil {
+		return "", fmt.Errorf("error fetching deposit address: %w", err)
+	}
+
+	return res.Data.Address, nil
+}
+
+func (c *coinbase) getTransaction(ctx context.Context, assetID uint32, txID string) (*cbtypes.Transaction, error) {
+	ticker, found := c.idTicker[assetID]
+	if !found {
+		return nil, fmt.Errorf("no ticker found for asset ID %d", assetID)
+	}
+
+	acctID, err := c.getAccountID(ctx, ticker)
+	if err != nil {
+		return nil, err
+	}
+
+	endpoint := fmt.Sprintf("v2/accounts/%s/transactions/%s", acctID, txID)
+	var res cbtypes.TransactionResponse
+	if err := c.request(ctx, http.MethodGet, endpoint, nil, nil, &res); err != nil {
+		return nil, fmt.Errorf("error fetching transaction: %w", err)
+	}
+
+	return res.Data, nil
+}
+
+// getTransactionByTxID returns a transaction by transaction ID, rather than
+// the coinbase ID. It will not return transactions older than 3 days.
+func (c *coinbase) getTransactionByTxID(ctx context.Context, assetID uint32, txID string) (*cbtypes.Transaction, error) {
+	ticker, found := c.idTicker[assetID]
+	if !found {
+		return nil, fmt.Errorf("no ticker found for asset ID %d", assetID)
+	}
+
+	acctID, err := c.getAccountID(ctx, ticker)
+	if err != nil {
+		return nil, err
+	}
+
+	nextURI := fmt.Sprintf("v2/accounts/%s/transactions", acctID)
+	for i := 0; i < 100 && nextURI != ""; i++ {
+		var res cbtypes.ListTransactionsResponse
+		if err := c.request(ctx, http.MethodGet, nextURI, nil, nil, &res); err != nil {
+			return nil, fmt.Errorf("error fetching transactions: %w", err)
+		}
+
+		for _, tx := range res.Data {
+			if tx.Network.Hash == txID {
+				return tx, nil
+			}
+		}
+
+		if len(res.Data) > 0 {
+			date := res.Data[len(res.Data)-1].CreatedAt
+			if date.Before(time.Now().Add(-time.Hour * 24 * 3)) {
+				return nil, fmt.Errorf("transaction %s not found", txID)
+			}
+		}
+
+		nextURI = res.Pagination.NextURI
+		nextURI = strings.TrimPrefix(nextURI, "/")
+	}
+
+	return nil, fmt.Errorf("transaction %s not found", txID)
+}
+
+// ConfirmDeposit checks if a deposit has been confirmed and returns the
+// amount deposited. This will only work for deposits that have been
+// in the past 3 days.
```

Comment:
```suggestion
// ConfirmDeposit checks if a deposit has been confirmed and returns the
// amount deposited. This will only work for deposits that have been made
// in the past 3 days.
```